Sidebar dev

zghera opened the pull request:
This PR adds on to #3 - Navbar Dev. The main addition is a sidebar that contains email, github, and linkedin profile logos with corresponding links to each. The main challenge here was the positioning on the side of the screen so that there was never overlap with the top navbar or any of the content. Another feature is the transition from a sidebar to a bottom ribbon/bar once the screen width is less than 600 px. This makes room for the content section while also providing immediate access to these contact links. 

Before:
![before](https://user-images.githubusercontent.com/39711376/83286521-0b0cef80-a1ae-11ea-989d-17800efd789d.png)

After (wide window):
![after-wide](https://user-images.githubusercontent.com/39711376/83286544-119b6700-a1ae-11ea-9f7d-148423e69a42.png)

After (narrow window): 
![after-narrow](https://user-images.githubusercontent.com/39711376/83286550-15c78480-a1ae-11ea-9337-4fce71a0ce9e.png)

zghera committed:
Merge branch 'sidebar-dev' of https://github.com/zghera/STEP-Portfolio into sidebar-dev

zghera committed:
initial sidebar progress

Able to get a sidebar that does not overlap with main content and test elements in sidebar switch to the bottom when the window width is reduced. Now need to figure out how to format the icons nicely once they transition to the bottom of the page.

zghera committed:
Working sidebar with flexboxgit add *

The sidebar container was changes so that while the container was not an element of a flexbox container, it was a flexbox itself (it contained flexbox elements). This enabled moving the logos to the bottom of the page when the window width was decreased to less than 600px.

zghera committed:
Minor color and syntax touch-ups

beiebi0 commented on portfolio/src/main/webapp/styles/style.css:
```diff
@@ -0,0 +1,149 @@
+ 
+/* Navigation Bar */
+#outter-nav {
+  top: 0;
+  width: 100vw;
+  position: fixed;
+  background-color: #02003C;
+  color: #55FBDC;
+  padding: 2vmin;
+  padding-top: 3vmin;
+ 
+  display: flex;
+  flex-flow: row wrap;
+  justify-content: space-between;
```
Please sort these to alphabetical order and fix the rest of the file . Re-https://google.github.io/styleguide/htmlcssguide.html#CSS_Formatting_Rules.

zghera replied:
Done.

zghera committed:
alphabatize CSS declarations (fix PR comment)

zghera committed:
(Hacky) fix of bottom/side bar in narrow view so it extends the full width

This was done by aking the width of the sidebar 110vw and moving it to the left by 5 vw to re-center the icons.

zghera committed:
Periods on sentences and 'outer' spelling

zghera committed:
Add project id to pom.xml

zghera committed:
Merge branch 'master' into sidebar-dev

zghera committed:
fix missing semi-colon in syle.css

zghera committed:
Merge branch 'sidebar-dev' of https://github.com/zghera/STEP-Portfolio into sidebar-dev

zghera committed:
Fix curly bracket styling

HiramSilvey commented on portfolio/src/main/webapp/styles/style.css:
```diff
@@ -1,9 +1,145 @@
-#content {
```
Could you sort the CSS selectors either by type (i.e. foo, .bar, #baz) or by relevant group? The current ordering is not very clear to me, and makes it harder to mentally parse.

zghera replied:
I will paste my most recent commit in response to this comment below. Please let me know if that makes sense to you or if you would like to discuss further. Thanks!

Commit 529ac25
> **Change order of CSS selectors for general elements and add comments for top navigation bar**
> 
> These changes were made to ensure readability with respect to the order of the CSS selectors. For general elements, the selectors were sorted by type from most general selector to most specific (element tag, class, id). For the rest of the CSS selectors, they were organized based on the associated HTML heirarchy. For example -- Top Navigation Bar: The first selector is the overall navigation bar container that has the id of 'outer-nav', the next two selectors are for the home logo and the right section of the nav-bar, as these are the child elements of the overall nav-bar. This recursive nature continues and is shared by the sidebar CSS selectors.

zghera committed:
Change order of CSS selectors for general elements and add comments for top navigation bar

These changes were made to ensure readability with respect to the order of the CSS selectors. For general elements, the selectors were sorted by type from most general selector to most specific (element tag, class, id). For the rest of the CSS selectors, they were organized based on the associated HTML heirarchy. For example -- Top Navigation Bar: The first selector is the overall navigation bar container that has the id of 'outer-nav', the next two selectors are for the home logo and the right section of the nav-bar, as these are the child elements of the overall nav-bar. This recursive nature continues and is shared by the sidebar CSS selectors.

zghera committed:
Merge branch 'master' into sidebar-dev

HiramSilvey commented on portfolio/src/main/webapp/index.html:
```diff
@@ -138,9 +155,9 @@ <h4>Robotic Systems Team Lead | Sep 2018 - May 2019</h4>
           <h2>Get In Touch</h4>
           <p>...</p>
-          <a href="mailto:zachary.ghera@gmail.com" id="link-box">Say Hi</a>
+          <a style="padding-bottom:20vh" href="mailto:zachary.ghera@gmail.com" id="link-box">Say Hi</a>
```
Is this "link-box" ID used anywhere? It's probably better to keep all styling together in the CSS file, rather than mixing between HTML and CSS files. You could use the link-box ID to apply the padding-bottom style in the CSS file, and remove the "style" attribute here. If you think it's better with the styling here, then remove the unused ID instead.

zghera replied:
See commit 6a6fd58.  tldr: removed this link entirely.

> Remove link to email in Contact section.
> 
> This change is in response to comment about an unused CSS id. This link was taken out in a latter branch commit as the sidebar already provides access to my email to contact me. A latter addition will include an arrow to the sidebar or pull the sidebar icons to the bottom regardless of the window side when the user scrolls to the 'Contact' page.

zghera committed:
Remove link to email in Contact section.

This change is in response to comment about an unused CSS id. This link was taken out in a latter branch commit as the sidebar already provides access to my email to contact me. A latter addition will include an arrow to the sidebar or pull the sidebar icons to the bottom regardless of the window side when the user scrolls to the 'Contact' page.
